### ISSUE TITLE
user_nl_cam silently ignored after cam->eam change

### ISSUE THREAD

**Issue by PeterCaldwell** [E3SM-Project/E3SM#3899]
Several scream folks (@ndkeen , @brhillman , @wlin7 )  report getting burnt after the cam->eam name change by using old run scripts which specify config options in a "user_nl_cam" file rather than a "user_nl_eam" file. Apparently all of your changes to user_nl_**cam** are silently ignored in runs using recent versions of master. It would be good for CIME to check if user_nl_cam is not empty for E3SM runs and to raise an exception if it is. Pinging @golaz who might care about this and @jgfouca who might know of a quick fix.

**Comment by whannah1** [comment]
I got bit by this too, but I would prefer to let people learn from their mistakes on this one. The only good way I can think to deal with this is just check for the "user_nl_cam" file and die with an error if it exists in the case directory. I wouldn't want to assume they just wrote to the wrong file and just combine the two files, or something like that.

**Comment by ndkeen** [comment]
Indeed I knew this would cause trouble and sure enough, I fell for it on a couple of occasions. I predict it will continue to cause trouble for folks as they use scripts. 

What about checking for the following: If user_nl_cam exists _and_ user_nl_eam does not, then fail or huge WARNING at setup/build time?

**Comment by PeterCaldwell** [comment]
Yeah, I totally agree with Walter that our fix shouldn't silently correct the incorrect name - it should die with an error. I hope that's what you meant by "learn from mistakes", @whannah1 ... letting them spend a day debugging why their runs look weird would probably be powerful teaching but feels a little too cruel ;-).

**Comment by whannah1** [comment]
😈

**Comment by maltrud** [comment]
i've also been bitten by this.  i disagree a bit with @ndkeen on failing only if user_nl_cam exists and user_nl_eam does not.  the latter always exists, whether it is modified or not.  so maybe just checking whether user_nl_cam exists.  and it should die, not warn.

**Comment by jgfouca** [comment]
@WesCoomber , this should be an easy one.

**Comment by beharrop** [comment]
I'm happy to be another statistic in this category.  I like @PeterCaldwell 's idea to have CIME issue an error and die if user_nl_cam exists in the directory.  My understanding is the only reason that file would exist is if the runscript is creating it.

**Comment by vanroekel** [comment]
Same for me.  I agree that the best solution is for CIME to error and fail if user_nl_cam exists in the directory.

**Comment by rljacob** [comment]
@wlin brought this up during the IG call today and we'll make it error out if user_nl_cam exists.  Same thing for user_nl_clm.  This only requires a mod to the buildnml in eam and elm, not the main cime source.  @wlin was going to do the eam change.

**Comment by tangq** [comment]
@rljacob , perhaps too late to bring up this. I didn't think about this at the beginning of the renaming of user_nl_cam/clm. But if we are going to work on them, I suggest renaming to user_nl_atm and user_nl_lnd.

The benefits are: 1) These are general names, like atm_in and lnd_in, independent of the model name (in case another model name change after the big day?) 2) Consistent naming conventions for user_nl_* and *_in. I understand if it's too late for changing the names too.

**Comment by PeterCaldwell** [comment]
Great, I'm glad this issue is getting traction. Thanks for the fix, @wlin7 ! I like @tangq 's idea of user_nl_atm if it is easy to do, but hopefully we will never have to change the name of the atm model again so maybe it isn't important?

**Comment by wlin7** [comment]
This issue is addressed by PR #3900.
